parallel-install/pkg/engine/engine.go

```diff
@@ -64,59 +71,57 @@ type Installation interface {
 	//However, once the underlying parallel operations end, the cancel condition is detected and the return channel is closed.
 	//All remaining components are not processed then.
 	//
-	Uninstall(ctx context.Context) (<-chan components.KymaComponent, error)
+	Uninstall(ctx context.Context) (<-chan components.KymaComponent, <-chan error, error)
 }
 
-func (e *Engine) Deploy(ctx context.Context) (<-chan components.KymaComponent, error) {
-
-	cmps, err := e.componentsProvider.GetComponents()
-	if err != nil {
-		return nil, err
-	}
-
+func (e *Engine) Deploy(ctx context.Context) (<-chan components.KymaComponent, <-chan error, error) {
 	//TODO: Size dependent on number of components?
 	statusChan := make(chan components.KymaComponent, 30)
+	errorChan := make(chan error)
 
-	//TODO: Can we avoid this goroutine? Maybe refactor run() so it's non-blocking ?
+	//TODO: Can we avoid this go-routine? Maybe refactor run() so it's non-blocking ?
```

Comment:
There is no misspelling here, golang's threads are called `goroutines`.
Reference: https://golang.org/doc/effective_go#goroutines

Reply:
Code spell complained and suggest go-routines :D ... will revert it

Reply:
FIXED